1/src/vig.rs

```diff
@@ -0,0 +1,69 @@
+use std::net::TcpStream;
+
+let nuid = 001763959;
+let hostname = "login.ccs.neu.edu";
+let port = 27993;
+
+fn main() -> {
+    let mut stream = TcpStream::connect(format!("{}:{}", hostname, port));
+    stream.write(format!("cs3700fall2015 HELLO {}", NUID));
+    loop {
+        // maybe a loop or something...?
+    }
+}
+
+struct Status {
+    message: String,
+    first: i32,
+    operator: String,
+    second: i32
+}
+
+impl Status {
+    fn make(&response_vec) -> Status {
```

Comment:
The `&response_vec` has no type because Rust is assuming that's what you're calling the `self` argument. Further you can't have a heterogeneous array.